windowing/gbm: fallback to non-alpha visual id

Kwiboo opened the pull request:
## Description
This PR fixes an issue starting kodi-gbm with libmali after #14515, ``XRGB8888`` is the expected  `EGL_NATIVE_VISUAL_ID` on mali utgard. Both alpha and non-alpha formats is now treated as valid config.

This needs to be tested with 10-bit + 8-bit fallback using Intel / mesa, ping @lrusak @a1rwulf

## Motivation and Context
kodi-gbm no longer starts using libmali on amlogic and rockchip devices.

```
DEBUG: failed to find matching EGL visual id
DEBUG: Previous line repeats 1 times.
FATAL: CApplication::Create: Unable to init windowing system
```

## How Has This Been Tested?
Kodi master starts on my Rock64 with this patch.

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

lrusak commented:
As discussed in private I'd rather not merge this as this is due to a bug in Mali not returning the correct EGL config for a given `EGL_NATIVE_VISUAL_ID`. We do not want to work around driver bugs in Kodi. So we can leave this as documentation as to why something may be broken. The reference to `we need to provide an alpha format to egl to workaround a mesa bug` wasn't actually a bug and was just a mistake on my part.

As for following correct EGL spec and using proper fallbacks we can rework that to be more obvious about we require and are actually using. I do not want to have references to the formats being used all over the place. This is why I changed it to reference from the plane itself what format is being used.

If you don't want to continue with this cleanup just say so and I can do some rework to make stuff more obvious. I also need to fix the fact that I broke the offscreenmodesetting because no visual id is set in that case.

I don't want to discourage you from contributing here. So feel free to do some of the rework if required (though we are quite time constrained due to the release period).

Kwiboo commented:
> I'd rather not merge this as this is due to a bug in Mali not returning the correct EGL config for a given EGL_NATIVE_VISUAL_ID. We do not want to work around driver bugs in Kodi. So we can leave this as documentation as to why something may be broken.

The specification violation with libmali would be that `eglCreatePlatformWindowSurfaceEXT` do not return `EGL_BAD_MATCH` when `EGL_NATIVE_VISUAL_ID` is `GBM_FORMAT_XRGB8888` and we have a `gbm_surface` that was created with `GBM_FORMAT_ARGB8888`. Technically it does not break the specification as we use `eglCreateWindowSurface` and not `eglCreatePlatformWindowSurfaceEXT` on libmali.

> I do not want to have references to the formats being used all over the place. This is why I changed it to reference from the plane itself what format is being used.

I have now reworked this without the need for another format variable, code should also be easier to follow as window/surface is created in a single method instead of being scattered across two classes. 

Currently the plane format variable is used for two different purposes, setting format capabilities of planes during init and secondary to switch between alpha/non-alpha for the drm framebuffer. I will try to cleanup this and some other smaller stuff that is bugging me in a day or two.

The main change this PR is introducing is adding a fallback to allow XRGB in addition to ARGB egl configs and enforce the EGL specification by using the `EGL_NATIVE_VISUAL_ID` when we create the `gbm_surface` instead of forcing argb format.

lrusak reviewed (approved):
Looks good other than the small change (I know it was like that before)

Kwiboo committed:
windowing/gbm: fallback to non-alpha visual id

Kwiboo committed:
windowing/gbm: use visual id to create gbm surface

Kwiboo committed:
windowing/gbm: move CreateNewWindow to WinSystemGbmEGLContext

Kwiboo commented on xbmc/windowing/gbm/WinSystemGbmEGLContext.cpp:
```diff
@@ -35,13 +32,18 @@ bool CWinSystemGbmEGLContext::InitWindowSystemEGL(EGLint renderableType, EGLint
   }
 
-  if (!m_eglContext.ChooseConfig(renderableType, visualId))
+  uint32_t visualId = m_DRM->GetOverlayPlane()->format;
```
I have now changed to use `GetFormat()`, I also changed the moved `CLog::Log` calls to use `{}` instead of `%s` in the update
